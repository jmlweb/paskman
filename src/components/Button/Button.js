import React from 'react';
import PT from 'prop-types';
import styled from 'styled-components';
import styledMap from 'styled-map';
import colors from '../../styles/colors';
import fonts from '../../styles/fonts';
import spacing from '../../styles/spacing';
import timings from '../../styles/timings';

const StyledBtnTemplate = styled.button`
  appearance: none;
  background: ${styledMap('color', {
    primary: colors.primary,
    secondary: colors.secondary,
    success: colors.success,
  })};
  border: 0;
  border-radius: 4px;
  color: ${styledMap('color', {
    primary: colors.secondary,
    secondary: colors.white,
    success: colors.white,
  })};
  font-family: ${fonts.fontFamily};
  font-size: ${styledMap('size', {
    xs: '1rem',
    sm: '1.2rem',
    md: '1.8rem',
    lg: '2rem',
    xl: '2.2rem',
  })};
  font-weight: 700;
  outline: 0;
  padding: ${styledMap('size', {
    xs: spacing.xs,
    sm: spacing.xs,
    md: spacing.sm,
    lg: spacing.md,
    xl: spacing.lg,
  })};
  text-transform: uppercase;
  transition: background 0.4s ${timings.easeOutQuad};
  width: ${(props) => {
    if (props.block) {
      return '100%';
    }
    return 'auto';
  }};
  &:active {
    background: ${styledMap('color', {
    primary: colors.primaryT2,
    secondary: colors.secondaryT2,
    success: colors.successT2,
  })};
  }
`;

const createStyledBtn = tag => StyledBtnTemplate.withComponent(tag);

export default function Button(props) {
  const {
    tag,
    color,
    size,
    block,
    children,
    ...rest
  } = props;
  const StyledBtn = createStyledBtn(tag);
  return <StyledBtn color={color} block={block} size={size} {...rest}>{children}</StyledBtn>;
}

Button.propTypes = {
  tag: PT.string,
  color: PT.string,
  size: PT.string,
  block: PT.bool,
  children: PT.string.isRequired,
};

Button.defaultProps = {
  tag: 'button',
  color: 'secondary',
  size: 'md',
  block: false,
};
