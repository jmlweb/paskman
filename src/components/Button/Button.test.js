import React from 'react';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Button from './Button';

configure({ adapter: new Adapter() });

describe('Button', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(<Button color="primary">Dummy</Button>);
  });
  it('Render the component', () => {
    expect(wrapper.find(Button).length).toEqual(1);
  });
  it('Matchs snapshot', () => {
    const tree = renderer.create(wrapper).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
