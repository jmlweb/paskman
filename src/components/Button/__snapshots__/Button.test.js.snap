// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Matchs snapshot 1`] = `
<button
  className="sc-bxivhb kSXfCd"
  color="primary"
  size="md"
>
  Dummy
</button>
`;
